com.unity.ml-agents/Runtime/Communicator/RpcCommunicator.cs

```diff
@@ -113,62 +114,75 @@ public UnityRLInitParameters Initialize(CommunicatorInitParameters initParameter
                     {
                         RlInitializationOutput = academyParameters
                     },
-                    out input);
-
-                var pythonPackageVersion = initializationInput.RlInitializationInput.PackageVersion;
-                var pythonCommunicationVersion = initializationInput.RlInitializationInput.CommunicationVersion;
-                var unityCommunicationVersion = initParameters.unityCommunicationVersion;
-
-                TrainingAnalytics.SetTrainerInformation(pythonPackageVersion, pythonCommunicationVersion);
-
-                var communicationIsCompatible = CheckCommunicationVersionsAreCompatible(unityCommunicationVersion,
-                    pythonCommunicationVersion,
-                    pythonPackageVersion);
-
-                // Initialization succeeded part-way. The most likely cause is a mismatch between the communicator
-                // API strings, so log an explicit warning if that's the case.
-                if (initializationInput != null && input == null)
+                    out input
+                );
+            }
+            catch (Exception ex)
```

Comment:
could you do cascading catches here instead of type checking? i.e.

```
catch (RpcException ex) {}
catch (AnotherException ex) {}
catch (Exception ex) { // unknown exceptions handled here }
```

Reply:
Discussed offline, this is to avoid duplicating the common code between RpcException and Exception. Updated this to use pattern matching. https://docs.microsoft.com/en-us/dotnet/csharp/pattern-matching